### PR TITLE
Basic implementation of Ecto.Adapter.Queryable

### PR DESCRIPTION
**Pull request by v0idpwn**
- Allows us to use Repo.get, Repo.all, Repo.one, Repo.reload through the "native" ecto interface
- Allows us to issue some simple queries and give errors when trying to use invalid queries

See tests for examples

### CODE REVIEW

File: lib/vax/adapter/query.ex [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        %{select: %{from: {:any, {:source, _source_schema, _, fields_and_types}}}} = _query_meta
+      ) do
+    Keyword.keys(fields_and_types)
+  end
+
+  defp where_expr_to_key({:or, _, exprs}, source, primary_key, bucket, params) do
+    Enum.map(exprs, &where_expr_to_key(&1, source, primary_key, bucket, params))
+  end
+
+  defp where_expr_to_key(
```
Comment (v0idpwn) [inline lib/vax/adapter/query.ex:30]: Add support for `in` expr.

Comment (v0idpwn) [reply lib/vax/adapter/query.ex:30]: I'm having a problem with dumping the parameter passed to the expr. I'll drop this feature for this PR and create an issue.

File: lib/vax/adapter.ex [hunk 2]
```diff
@@ -6,10 +6,44 @@ defmodule Vax.Adapter do
...
+
+    execute_static_transaction(adapter_meta.repo, fn conn, tx_id ->
+      {:ok, results} = :antidotec_pb.read_objects(conn, objs, tx_id)
+
+      results =
+        for result <- results,
+            result_value = :antidotec_map.value(result),
+            result_value != %{} do
+          map = Map.new(result_value, fn {{k, _t}, v} -> {String.to_atom(k), v} end)
+          Enum.map(fields, &Map.get(map, &1))
```
Comment (v0idpwn) [inline lib/vax/adapter.ex:41]: This looks a bit weird, but I think it is a reasonably performative way to turn the base `%{{key, type} => value}` map into the ordered list of fields.

File: lib/vax/adapter/query.ex [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  def query_to_objs(query, params, bucket) do
+    {source, schema} = query.from.source
+    primary_key = Vax.Adapter.Helpers.schema_primary_key!(schema)
+
+    case query.wheres do
+      [%{expr: expr}] ->
+        where_expr_to_key(expr, source, primary_key, bucket, params) |> List.flatten()
+
+      _ ->
+        raise "Vax only supports a single `where` expression per query"
```
Comment (v0idpwn) [inline lib/vax/adapter/query.ex:11]:
```suggestion
      wheres ->
        raise "Vax only supports a single `where` expression per query. Received #{Enum.count(wheres)}"
```

File: lib/vax/adapter.ex (hunk 2 above)
Comment (v0idpwn) [reply lib/vax/adapter.ex:41]: Accepting any tips :D

File: lib/vax/adapter/query.ex (hunk 1 above)
Comment (v0idpwn) [reply lib/vax/adapter/query.ex:30]: Actually, just solved that

File: lib/vax/adapter.ex [hunk 4]
```diff
@@ -5,11 +5,46 @@ defmodule Vax.Adapter do
...
 
-  @bucket "vax"
+  @impl Ecto.Adapter.Queryable
+  def stream(_adapter_meta, _query_meta, _query_cache, _params, _options) do
+    raise "Not implemented"
+  end
+
+  @impl Ecto.Adapter.Queryable
+  def prepare(:update_all, _query), do: raise("Not implemented")
+  def prepare(:delete_all, _query), do: raise("Not implemented")
```
Comment (v0idpwn) [inline lib/vax/adapter.ex:23]: I don't think these are useful for Vaxine at this point

**Review by icehaunter — APPROVED** [review]

**Pull request merged**